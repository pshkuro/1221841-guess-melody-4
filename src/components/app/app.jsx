import PropTypes from 'prop-types';
import React from "react";
import WelcomeScreen from "../welcome-screen/welcome-screen.jsx";

export default function App(props) {
  const {errorsCount} = props;

  return (
    <WelcomeScreen errorsCount={errorsCount}/>
  );
}

App.propTypes = {
  errorsCount: PropTypes.number.isRequired,
};

