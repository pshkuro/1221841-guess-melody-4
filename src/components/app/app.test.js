import {App} from "./app.jsx";
import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);


const props = {
  questions: [
    {
      type: `genre`,
      genre: `rock`,
      answers: [{
        src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
        genre: `rock`,
        id: 1212,
      }, {
        src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
        genre: `blues`,
        id: 89,
      }, {
        src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
        genre: `jazz`,
        id: 343434,
      }, {
        src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
        genre: `rock`,
        id: 1212,
      }],
    }, {
      type: `artist`,
      song: {
        artist: `Jim Beam`,
        src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      },
      answers: [{
        picture: `https://api.adorable.io/avatars/128/1`,
        artist: `John Snow`,
        id: 9090,
      }, {
        picture: `https://api.adorable.io/avatars/128/2`,
        artist: `Jack Daniels`,
        id: 43444,
      }, {
        picture: `https://api.adorable.io/avatars/128/3`,
        artist: `Jim Beam`,
        id: 908,
      }],
    },
  ],
  errorsCount: 3,
  onUserAnswer: jest.fn(),
  onWelcomeButtonClick: jest.fn,

};


describe(`Render App`, () => {
  it(`Render WelcomeScreen`, () => {
    const store = mockStore({
      mistakes: 0,
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <App
              {...props}
              step={-1}
            />
          </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render GenreQuestionScreen`, () => {
    const store = mockStore({
      mistakes: 3,
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <App
              {...props}
              step={0}
            />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render ArtistQuestionScreen`, () => {
    const store = mockStore({
      mistakes: 3,
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <App
              {...props}
              step={-1}
            />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
