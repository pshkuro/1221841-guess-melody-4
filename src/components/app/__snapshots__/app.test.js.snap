// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Render ArtistQuestionScreen 1`] = `
<section
  className="welcome"
>
  <div
    className="welcome__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <button
    className="welcome__button"
    onClick={[Function]}
  >
    <span
      className="visually-hidden"
    >
      Начать игру
    </span>
  </button>
  <h2
    className="welcome__rules-title"
  >
    Правила игры
  </h2>
  <p
    className="welcome__text"
  >
    Правила просты:
  </p>
  <ul
    className="welcome__rules-list"
  >
    <li>
      Нужно ответить на все вопросы.
    </li>
    <li>
      Можно допустить 
      3
       ошибки.
    </li>
  </ul>
  <p
    className="welcome__text"
  >
    Удачи!
  </p>
</section>
`;

exports[`Render App Render GenreQuestionScreen 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите 
      rock
       треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-1212"
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-1212"
          />
          <label
            className="game__check"
            htmlFor="answer-1212"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-89"
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-89"
          />
          <label
            className="game__check"
            htmlFor="answer-89"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-343434"
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-343434"
          />
          <label
            className="game__check"
            htmlFor="answer-343434"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-1212"
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-1212"
          />
          <label
            className="game__check"
            htmlFor="answer-1212"
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;

exports[`Render App Render WelcomeScreen 1`] = `
<section
  className="welcome"
>
  <div
    className="welcome__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <button
    className="welcome__button"
    onClick={[Function]}
  >
    <span
      className="visually-hidden"
    >
      Начать игру
    </span>
  </button>
  <h2
    className="welcome__rules-title"
  >
    Правила игры
  </h2>
  <p
    className="welcome__text"
  >
    Правила просты:
  </p>
  <ul
    className="welcome__rules-list"
  >
    <li>
      Нужно ответить на все вопросы.
    </li>
    <li>
      Можно допустить 
      3
       ошибки.
    </li>
  </ul>
  <p
    className="welcome__text"
  >
    Удачи!
  </p>
</section>
`;
